src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -304,123 +323,150 @@
+                                                    y)
+               ;; NOTE: when the keyboard is shown and the message list
+               ;; doesn't fill the screen, we use a different value for
+               ;; scroll-y, which would make sure the avatar is minimized
+               ;; appropriately and the header is shown.
+               keyboard-shown-with-unfilled-list? (< y window-height)
+               keyboard-space                     (- window-height y)
+               keyboard-scroll-value              (- y keyboard-space keyboard-space)
+               ;; NOTE: doing the calculations as we would in `on-scroll`, so the animations
+               ;; and interpolations are triggered properly.
+               content-size                       (- y
+                                                     window-height
+                                                     ;; NOTE: for some reason android insets are
+                                                     ;; included in the content size `y`, while iOS
+                                                     ;; are not.
+                                                     (when platform/android? (:top insets)))
+               current-y                          (max (- content-size-shared
+                                                          scroll-y-shared)
+                                                       0)
+               new-scroll-value                   (if keyboard-shown-with-unfilled-list?
+                                                    keyboard-scroll-value
+                                                    (- content-size
+                                                       current-y))]
+           (when (and (>= new-scroll-value 0)
+                      (or (= scroll-y-shared 0)
+                          (> (Math/abs (- content-size-shared y))
+                             min-message-height)))
+             (reset! messages-scroll-y-value-initialized? true)
+             (scroll-handler new-scroll-value content-size))))
+       :on-end-reached #(list-on-end-reached scroll-y on-end-reached?)
+       :on-scroll-to-index-failed identity
+       :scroll-indicator-insets {:top (if (:able-to-send-message? context)
+                                        (- composer.constants/composer-default-height 16)
+                                        0)}
+       :keyboard-dismiss-mode :interactive
+       :keyboard-should-persist-taps :always
+       :on-scroll-begin-drag #(do
+                                (rf/dispatch [:chat.ui/set-input-focused false])
+                                (rn/dismiss-keyboard!))
+       :on-momentum-scroll-begin state/start-scrolling
+       :on-momentum-scroll-end state/stop-scrolling
+       :scroll-event-throttle 16
+       :on-scroll (fn [event]
+                    (let [content-size-y
+                          (- (oops/oget event "nativeEvent.contentSize.height")
```

Comment:
btw, please can you elaborate this change?
This was already a separate function, why we are moving it here? (this will unnecessary increase the size of flat list function. Which is already quite big and harder to maintain.

Reply:
So there's the `scroll-y` animated value that is used for interpolating the header avatar and navigation animations. It is mainly set during `on-scroll`, but we need to control it from `on-content-size-change` as well because `on-scroll` is not called on the initial render of the screen, when messages are incoming and when the keyboard is shown. Since the logic for showing/hiding the navigation header was in that `on-scroll` fn, I had to abstract it into a separate function that is used by both `on-scroll` and `on-content-size-change` and what was added here is what is uncommon. I could extract it into another function if that helps readability.